doc/nrf/device_guides/working_with_nrf/nrf53/qspi_xip_guide.rst

```diff
@@ -0,0 +1,321 @@
+* :kconfig:option:`CONFIG_HAVE_CUSTOM_LINKER_SCRIPT`
+* :kconfig:option:`CONFIG_BUILD_NO_GAP_FILL`
+* :kconfig:option:`CONFIG_XIP`
+* :kconfig:option:`CONFIG_NORDIC_QSPI_NOR_XIP`
+* :kconfig:option:`CONFIG_XIP_SPLIT_IMAGE`
+* :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` - This option enables the MCUboot build and support for the application.
+
+Additionally set the following options:
+
+* :kconfig:option:`CONFIG_CUSTOM_LINKER_SCRIPT` to ``"<linker_file_for_relocation>"``
+* :kconfig:option:`CONFIG_UPDATEABLE_IMAGE_NUMBER` to ``3``
+* :kconfig:option:`CONFIG_FLASH_INIT_PRIORITY` to ``40`` - It is required to ensure the QSPI device initialization priority, which makes the external XIP code accessible before it is executed.
+  If any initialization code is expected to be run from the QSPI XIP, then its initialization priority value must be lower than the QSPI device initialization priority.
+
+Setting up QSPI flash
+*********************
+
+The QSPI flash DTS must be available for the application and the MCUboot.
+You must correctly set up the QSPI flash chip in the board devicetree file, including the operating mode.
+The flash chip does not have to run in the QSPI mode for XIP to function, but using other modes will reduce the execution speed of the application.
+
+See the following snippet for an example of the Nordic Thingy:53 configuration that supports DSPI:
+
+.. code-block:: devicetree
+
+        &qspi {
+        status = "okay";
+        pinctrl-0 = <&qspi_default>;
+        pinctrl-1 = <&qspi_sleep>;
+        pinctrl-names = "default", "sleep";
+        mx25r64: mx25r6435f@0 {
+            compatible = "nordic,qspi-nor";
+            reg = <0>;
+            writeoc = "pp2o";
+            readoc = "read2io";
+            sck-frequency = <8000000>;
+            jedec-id = [c2 28 17];
+            sfdp-bfp = [
+                e5 20 f1 ff  ff ff ff 03  44 eb 08 6b  08 3b 04 bb
+                ee ff ff ff  ff ff 00 ff  ff ff 00 ff  0c 20 0f 52
+                10 d8 00 ff  23 72 f5 00  82 ed 04 cc  44 83 68 44
+                30 b0 30 b0  f7 c4 d5 5c  00 be 29 ff  f0 d0 ff ff
+            ];
+            size = <67108864>;
+            has-dpd;
+            t-enter-dpd = <10000>;
+            t-exit-dpd = <35000>;
+        };
+    };
+
+.. note::
+    Due to QSPI peripheral product anomaly, the QSPI peripheral must be ran with the ``HFCLK192MCTRL=0`` setting.
+    Setting this to any other value may cause undefined operation of the device.
+
+Add the following to the DTS overlay for your board:
+
+.. code-block:: devicetree
+
+    / {
+        chosen {
+                nordic,pm-ext-flash = &mx25r64;
+        };
+    };
+
+Setting up static partition manager
+***********************************
+
+You need to complete the setup in order to use a static partitioning in your project.
+The configuration must have 3 images with 2 slots each:
+
+* The first set of slots is for the internal flash part of the application.
```

Comment:
For the future (not in this PR), perhaps we can make a figure that demonstrates this. If so, I can draft one and we can add it in later.

Reply:
That sounds like a great addition, agreed